sds/src/secondary_validation/jwt_expiration_checker.rs

```diff
@@ -0,0 +1,130 @@
+use crate::secondary_validation::Validator;
+use base64::{engine::general_purpose::URL_SAFE_NO_PAD, Engine};
+use chrono::Utc;
+use serde_json::Value as JsonValue;
+
+pub struct JwtExpirationChecker;
+
+const SEGMENTS_COUNT: usize = 3;
+
+impl Validator for JwtExpirationChecker {
+    fn is_valid_match(&self, regex_match: &str) -> bool {
+        let result = decode_segments(regex_match);
+        if let Ok((_, payload)) = result {
```

Comment:
Nit: I think a more idiomatic way to write this in rust is to inline directly the variable
```suggestion
        if let Ok((_, payload)) = decode_segments(regex_match) {
```